Sami/Parser/ClassVisitor/PropertyClassVisitor.php

```diff
@@ -50,58 +49,26 @@ public function visit(ClassReflection $class)
      * Adds a new property to the class using an array of tokens.
      *
      * @param ClassReflection $class       Class reflection
-     * @param string          $propertyTag Property tag contents
+     * @param string|array    $propertyTag Property tag contents
      *
      * @return bool
      */
     protected function injectProperty(ClassReflection $class, $propertyTag)
     {
-        if (!$data = $this->parseProperty($propertyTag)) {
-            return false;
-        }
-
-        $property = new PropertyReflection($data['name'], $class->getLine());
-        $property->setDocComment($data['description']);
-        $property->setShortDesc($data['description']);
-
-        if (isset($data['hint'])) {
-            $property->setHint(array(array($data['hint'], null)));
-        }
-
-        $class->addProperty($property);
-
-        return true;
-    }
-
-    /**
-     * Parses the parts of an @property tag into an associative array.
-     *
-     * @param string $tag Property tag contents
-     *
-     * @return array
-     */
-    protected function parseProperty($tag)
-    {
-        // Account for default array syntax
-        $tag = str_replace('array()', 'array', $tag);
-
-        /** @var PropertyTag $propertyTag */
-        $propertyTag = $this->context->getDocBlockParser()->getTag('@property '.$tag);
-        $propertyName = $propertyTag->getVariableName();
+        if (is_array($propertyTag) && count($propertyTag) == 3 && !empty($propertyTag[1])) {
```

Comment:
> Its is Sami class or phpDocumentor one?

The Sami class.

> There was a bug, that short description was partially lost. Probably there is no test currently that checks that it's not lost.

`@property` tags don't have a short description, they only consist of type, name and description. Or did I misunderstand you here?

> If you don't change injectProperty method, then all would work anyway?

No, since they now come properly parsed out of `ClassReflection::getTags()`, not as a single string like before.